gnomad_constraint/plots/plotting_functions.R

```diff
@@ -0,0 +1,374 @@
+    roc1,
+    roc2,
+    version1,
+    version2,
+    title_label,
+    color1 = "darkorange1",
+    color2 = "darkorchid3") {
+  # Combine two ROC plots into a single plot
+  # roc1: ROC object from pROC package
+  # roc2: ROC object from pROC package
+  # version1: Version label for ROC1
+  # version2: Version label for ROC2
+  # title_label: Title for the plot
+  # color1: Color for ROC1
+  # color2: Color for ROC2
+  # Returns: ggplot object
+
+  # Combine ROC outputs
+  roc_data1 <- as.data.frame(cbind(roc1$sensitivities, roc1$specificities))
+  roc_data2 <- as.data.frame(cbind(roc2$sensitivities, roc2$specificities))
+  roc_data1$version <- version1
+  roc_data2$version <- version2
+  all_rocs <- rbind(roc_data1, roc_data2)
+  colnames(all_rocs) <- c("sensitivity", "specificity", "version")
+
+  # Define AUC labels
+  auc1 <- paste("AUC: ", round(roc1$auc[1], 2), sep = "")
+  auc2 <- paste("AUC: ", round(roc2$auc[1], 2), sep = "")
+
+  # Plot ROC output
+  p <- ggplot(
+    all_rocs,
+    aes(1 - .data$specificity, .data$sensitivity, color = version)
+  ) +
+    geom_line(linewidth = 1, alpha = 0.9) +
+    theme_classic() +
+    theme(
+      axis.title = element_text(colour = "black", size = 12, face = "bold"),
+      axis.text = element_text(colour = "black", size = 10)
+    ) +
+    scale_color_manual(values = c(color1, color2)) +
+    labs(
+      x = "False Positive Rate (1-Specificity)",
+      y = "True Positive Rate (Sensitivity)",
+      color = "Version",
+      title = title_label
+    ) +
+    annotate("text", x = .70, y = .25, label = auc1, color = color1) +
+    annotate("text", x = .70, y = .20, label = auc2, color = color2)
+
+  return(p)
+}
+
+expected_projections <- function(
+    df,
+    label = "pLoF",
+    sample_size_df = dataset_sample_sizes,
+    xlimits = c(100, 1e8)) {
+  # Plot displaying the percent of genes that would be expected to have a
+  # certain number or variants based on sample size
+  # df: Dataframe with columns 'n_variants', 'n_required', and 'rank'
+  # label: Text that will be included at the top of the plot
+  # sample_size_df: Dataframe with columns 'intercepts', 'linetype', and
+  # 'label' for vertical lines on the plot that show the sample size of some
+  # key datasets
+  # xlimits: Define the limits of the x-axis
+  # Returns: ggplot object
+  df <- df %>%
+    mutate(
+      n_variants = forcats::fct_reorder(as.factor(.data$n_variants), .data$n_variants)
+    )
+
+  p <- ggplot(df, aes(y = .data$rank, x = .data$n_required, color = .data$n_variants)) +
+    geom_line(linewidth = 2) +
+    theme_classic() +
+    scale_x_log10(labels = comma, limits = xlimits) +
+    xlab("Sample size required") +
+    ylab("Percent of human genes") +
+    geom_vline(
+      data = sample_size_df,
+      aes(xintercept = .data$intercepts, linetype = .data$linetype),
+      key_glyph = "path"
+    )
+
+  # Add manual linetype scale for legend
+  p <- p +
+    scale_linetype_manual(
```

Comment:
the order of lines indicating the size of each database are no longer in the correct order

Reply:
oops, fixed I think